src/Adornment/Adornment.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ReactNode, cloneElement, RefObject } from 'react'
+
+import { AdornmentWrapper, StyledAdornment } from './styled'
+import { useRefDependentCssLengthValue } from '../useRefDependentCssLengthValue'
+import { isReactElement } from '../utils/isReactElement'
+
+export enum Position {
+  LEFT = 'Left',
+  RIGHT = 'Right'
+}
+
+type AdornmentProps = {
+  children: ReactNode
+
+  leftComponent?: ReactNode
+  rightComponent?: ReactNode
+  inset?: number
+}
+
+function tryGetRectWidth (ref: RefObject<HTMLElement>) {
+  if (ref.current) {
+    return ref.current.getBoundingClientRect().width
+  }
+}
+
+export const Adornment = ({
+  leftComponent,
+  rightComponent,
+  children,
+  inset = 10
+}: AdornmentProps) => {
+
+  const [paddingLeft, leftRef] = useRefDependentCssLengthValue({ baseValue: inset, refAccessor: tryGetRectWidth })
+  const [paddingRight, rightRef] = useRefDependentCssLengthValue({ baseValue: inset, refAccessor: tryGetRectWidth })
+
```

Comment:
Building on useClientRect() you could replace following functions with:
```
const [leftRect, leftRef] = useClientRect() 
const [rightRect, tightRef] = useClientRect() 

const paddingLeft = leftRect ? (leftRect + inset) : 0
const paddingRight = rightRect ? (rightRect + inset) : 0
```

Maybe im missing what useRefDependentCssLengthValue does exactly but in that case please explain, to me its not very self-explanatory.